csharp/LogicalBatchReader/ArrayReader.cs

```diff
@@ -140,6 +140,40 @@ public bool HasNext()
             return !_bufferedReader.IsEofDefinition;
         }
 
+        public long Skip(long numRowsToSkip)
+        {
+            for (var i = 0; i < numRowsToSkip; ++i)
+            {
+                if (_bufferedReader.IsEofDefinition)
+                {
+                    return i;
+                }
+
+                var defn = _bufferedReader.GetCurrentDefinition();
+                if (defn.DefLevel > _definitionLevel)
+                {
+                    if (typeof(TItem) == typeof(TLogical))
+                    {
+                        ReadLogicalTypeArray();
```

Comment:
It will probably require a bit more duplication of logic but I think it's worth optimizing this further by adding `SkipLogicalTypeArray` and `SkipInnerTypeArray` methods that have similar logic to the read methods but avoid creating arrays and copying values.

Reply:
Ok!